cdap-ui/app/hydrator/services/create/actions/available-plugins-action-creator.js

```diff
@@ -129,12 +129,14 @@ class PipelineAvailablePluginsActions {
     this.api.fetchAllPluginsProperties({ namespace }, reqBody)
       .$promise
       .then((res) => {
-        res.forEach((plugin, index) => {
+        res.forEach((plugin) => {
           let pluginProperties = Object.keys(plugin.properties);
           if (pluginProperties.length === 0) { return; }
 
           let pluginKey = pluginProperties[0].split('.')[1];
-          let key = plugins[index].key;
+          const { key } = plugins.find(pl =>
```

Comment:
Can there be a large number of plugins? This could run in `O(n^2)` in a bad case.

Reply:
You're right, optimized the implementation here.